src/Importer/ModuleImporter.ts

```diff
@@ -116,4 +108,66 @@ export class ModuleImporter implements ImporterContract {
       })
       .join('\n');
   }
+
+  protected transformScript(contents: string, resolveDir: string, sourcefile: string): string {
+    const { outputFiles } = buildSync({
+      stdin: {
+        contents,
+        resolveDir,
+        sourcefile,
+        loader: 'ts',
+      },
+      platform: 'node',
+      format: 'cjs',
+      external: ['apply'],
+      bundle: true,
+      write: false,
+    });
+
+    return outputFiles[0].text;
+  }
+
+  protected createContext(directory: string, filename: string): Record<string, any> {
+    const exports = {};
+    const moduleGlobals = {
+      exports,
+      require,
+      module: {
+        exports,
+        filename,
+        id: filename,
+        path: directory,
+        require: module.require,
+      },
+      __dirname: directory,
+      __filename: filename,
+    };
+
+    const nodeGlobals = {
+      Buffer,
+      clearImmediate,
+      clearInterval,
+      clearTimeout,
+      console,
+      global,
+      process,
+      queueMicrotask,
+      setImmediate,
+      setInterval,
+      setTimeout,
+      TextDecoder: (global as any).TextDecorator,
+      TextEncoder: (global as any).TextEncoder,
+      URL: (global as any).URL,
+      URLSearchParams: (global as any).URLSearchParams,
+      WebAssembly: (global as any).WebAssembly,
```

Comment:
I don't understand this. `global` will never be something else than `NodeJS.Global`, right? So, `TextDecorator`, `TextEncoder`, `URL`, `URLSearchParams` and `WebAssembly` will always be `undefined` inside the context, thus not solving the issue?

Reply:
These 5 globals are not defined in `NodeJS.Global` but it's only a TypeScript issue.
They are available in `nodejs >= 12`. You can test it on your computer using a simple repl.

```bash
$ node
Welcome to Node.js v12.22.1.
Type ".help" for more information.
> global.TextDecoder
[class TextDecoder]
>
```

I added the `global.[...]` to ensure that it will not throw on previous versions of node. If one is unavailable on a specific version it will fallback to undefined (as expected).

Reply:
Makes sense, thanks for the clarification!